nautobot/extras/tests/test_datasources.py

```diff
@@ -662,3 +685,29 @@ def populate_repo(path, url):
                 device = Device.objects.get(name=self.device.name)
                 self.assertIsNone(device.local_context_data)
                 self.assertIsNone(device.local_context_data_owner)
+
+    def test_git_dry_run(self, MockGitRepo):
+        with tempfile.TemporaryDirectory() as tempdir:
+            with self.settings(GIT_ROOT=tempdir):
+
+                def create_empty_repo(path, url, clone_initially=False):
+                    os.makedirs(path, exist_ok=True)
+                    return mock.DEFAULT
+
+                MockGitRepo.side_effect = create_empty_repo
+                MockGitRepo.return_value.checkout.return_value = self.COMMIT_HEXSHA
+
+                self.dummy_request.id = uuid.uuid4()
+                self.job_result = JobResult.objects.create(
+                    name=self.repo.name,
+                    obj_type=ContentType.objects.get_for_model(GitRepository),
+                    job_id=uuid.uuid4(),
+                )
+
+                git_repository_diff_origin_and_local(self.repo.pk, self.dummy_request, self.job_result.pk)
+                self.job_result.refresh_from_db()
+
+                self.assertEqual(self.job_result.status, JobResultStatusChoices.STATUS_COMPLETED, self.job_result.data)
+
+                MockGitRepo.return_value.checkout.assert_not_called()
+                MockGitRepo.return_value.diff_remote.assert_called()
```

Comment:
I'd like to see at least a couple more test cases, probably in a new `nautobot/utilities/tests/test_git.py` file:

1) Directly call `GitRepo(path, url, clone_initially=False) and verify that `Repo.clone_from()` was not called in this case
2) Directly call `GitRepo.diff_remote()` with a mocked (no-op) `self.fetch()` and a mocked (returning plausible Git CLI output) `self.repo.git.diff()` to verify that it returns the expected list of data.